index.js

```diff
@@ -582,6 +598,51 @@ export class WACZ {
     }
   }
 
+  /**
+   * Copies pages.jsonl and extraPages.jsonl files in this.pagesDir into ZIP.
+   * @returns {Promise<void>}
+   */
+  copyPagesFilesToZip = async () => {
+    this.stateCheck()
+
+    const { pagesDir, log, addFileToZip } = this
+
+    if (!pagesDir) {
+      throw new Error('Error copying pages files, no directory specified.')
+    }
+
+    const pagesFiles = await fs.readdir(pagesDir)
+
+    for (let i = 0; i < pagesFiles.length; i++) {
+      const filename = pagesFiles[i]
+      const filenameLower = filename.toLowerCase()
+      const pagesFile = resolve(this.pagesDir, filename)
+
+      if (!filenameLower.endsWith('.jsonl')) {
+        log.warn(`Pages: Skipping file ${pagesFile}, does not end with jsonl extension`)
+        continue
+      }
+
+      let isValidJSONL = true
+
+      // Ensure file is valid JSONL
+      const rl = readline.createInterface({ input: createReadStream(pagesFile) })
+      for await (const line of rl) {
+        try {
+          JSON.parse(line)
```

Comment:
Thanks for the edits @tw4l! 

I think we're almost there - We should maybe we add a little bit of [validation against the spec](https://specs.webrecorder.net/wacz/1.1.1/#pages-jsonl), just to make sure those are indeed pages files.

We could check: 
- That the first item contains `format` and `id` 
- That subsequent entries contain `url` and `ts`  

Maybe using Node's `assert` since we're in a try / catch ?

What do you think?

Reply:
Yeah nice suggestion! May as well get this right while we're focused on it :)

Reply:
Commit pushed!

Reply:
Ilya raised a question: in older versions of the crawler, the pages files occasionally included invalid lines, we think because of text extraction that wasn't truncated.

It may be safer if less performant to filter per-line rather than per-file, but write valid lines as-is into the correct file in the WACZ. What do you think?

Reply:
I'm now inclined to say that we can generate valid pages.jsonl and just fail if its invalid, but yeah, another option would be, on first failure, to skip the failing lines and reserialize only valid ones (similar to old behavior). But not sure if its needed at this point.

Reply:
I think either approach works for me; in both cases, we are unlikely to end up with invalid `pages.jsonl` files added to the archive. I am also not concerned about performance for this step.

Let me know if you'd like to add line-by-line filtering or not 😄 . Otherwise: this is great and I'm happy to test / approve / merge.

Thank you both!

Reply:
Sounds like we're going to stick with per-file and just make sure the crawler isn't writing any invalid pages files to begin with, so feel free to test/approve/merge, thank you!